spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/partitioner/TokenRangeSplitter.scala

```diff
@@ -1,16 +1,33 @@
 package com.datastax.spark.connector.rdd.partitioner
 
-import com.datastax.spark.connector.rdd.partitioner.dht.{Token, TokenRange}
-
-/** Splits a token range into smaller sub-ranges,
-  * each with the desired approximate number of rows. */
-trait TokenRangeSplitter[V, T <: Token[V]] {
-
-  /** Splits given token range into n equal sub-ranges. */
-  def split(range: TokenRange[V, T], splitSize: Long): Seq[TokenRange[V, T]]
-}
+import com.datastax.spark.connector.rdd.partitioner.dht.{Token, TokenFactory, TokenRange}
 
+import scala.collection.parallel.ForkJoinTaskSupport
+import scala.concurrent.forkjoin.ForkJoinPool
 
 
+/** Splits a token ranges into smaller sub-ranges,
+  * each with the desired approximate number of rows. */
+private[partitioner] trait TokenRangeSplitter[V, T <: Token[V]] {
+
+  def split(tokenRanges: Iterable[TokenRange[V, T]], splitCount: Int): Iterable[TokenRange[V, T]] = {
+    val wholeRing = tokenRanges.map(_.ringFraction).sum
```

Comment:
That's a great question! I'm not 100% sure the answer :) 

Apparently there was a [case found](https://datastax.slack.com/archives/drivers/p1465394335000262) when `Metadata#getTokenRanges` did not return the whole ring. Don't know if this was confirmed.

[Tried to ask](https://datastax.slack.com/archives/drivers/p1465821536000002) with no luck. Going to do that again.